File: Scala/day12.scala
class Student(first: String, last: String, ph: String) {
  val firstName: String = first
  val lastName: String = last
  val phone: String = ph

  def display() {
    println("First Name: " + firstName)
    println("Last Name: " + lastName)
    println("Phone: " + phone)
  }
}

class Grade(first: String, last: String, ph: String, sc: Int) extends Student(first, last, ph) {
  val score: Int = sc

  // Write your code here
  def calculate(): String = score match {
    case d if score < 40 => "D"
    case b if score >= 40 && score < 60 => "B"
    case a if score >= 60 && score < 75 => "A"
    case e if score >= 75 && score < 90 => "E"
    case _ => "O"
  }
}

object Solution {
  def main(args: Array[String]) {
    val sc = new java.util.Scanner(System.in)
    val (firstName, lastName, phone) = (sc.next(), sc.next(), sc.next())
    val score = sc.nextInt()
    val stu = new Grade(firstName, lastName, phone, score)
    stu.display()
    println("Grade: " + stu.calculate)
  }
}

File: Python/day12.py
class Student:
    def __init__(self, firstName, lastName, phone):
        self.firstName = firstName
        self.lastName = lastName
        self.phone = phone
    def display(self):
        print "First Name:", self.firstName
        print "Last Name:", self.lastName
        print "Phone:", self.phone

class Grade(Student):
    def __init__(self, firstName, lastName, phone, score):
        Student.__init__(self, firstName, lastName, phone)
        self.score = score

    # Write your code here
    def calculate(self):
        if self.score < 40:
            return "D"
        elif self.score >= 40 and self.score < 60:
            return "B"
        elif self.score >= 60 and self.score < 75:
            return "A"
        elif self.score >= 75 and self.score < 90:
            return "E"
        elif self.score >= 90 and self.score <= 100:
            return "O"

firstName = raw_input().strip()
lastName = raw_input().strip()
phone = int(raw_input())
score = int(raw_input())
stu = Grade(firstName, lastName, phone, score)
stu.display()
print "Grade:", stu.calculate()